R/editing.R

```diff
@@ -367,6 +367,15 @@ publish_update <- function(mn,
     # Get the metadata doc
     message("Getting metadata from the MN.")
     eml <- EML::read_eml(rawToChar(dataone::getObject(mn, metadata_pid)), asText = TRUE)
+
+  } else if(class(metadata_path) == "eml") {
+      # If an eml object is provided, use it directly after validating
+      if(!eml_validate(metadata_path)){
```

Comment:
@isteves Maybe move this out of the call here so that eml always gets validated wether or not is an R object or from disk. Also maybe add a message so users know that eml is being validated in case it takes a non-trivial amount of time.